asynch/proto/streams/buffered.py

```diff
@@ -111,31 +111,47 @@ def __init__(self, reader: StreamReader, buffer_max_size: int = constants.BUFFER
         self.current_buffer_size = 0
         self.position = 0
 
+    def _reset_buffer(self):
+        self.position = 0
+        self.buffer = bytearray()
+
     async def _read_into_buffer(self):
         packet = await self.reader.read(self.buffer_max_size)
         self.buffer.extend(packet)
         self.current_buffer_size = len(self.buffer)
 
+    async def _update_buffer_if_full(self):
+        if self.position == self.current_buffer_size:
+            self._reset_buffer()
+            await self._read_into_buffer()
+
     def _read_one(self):
+        if not self.buffer and not self.position:
+            return b""
```

Comment:
The case for an empty buffer when it is initialised or no data were read. The `self.buffer = b""` and the `self.position = 0` that will cause IndexError when executing the `packet = self.buffer[self.position]` statement.

Reply:
This is wrong, removed and the function is intact again.